akka-http/src/main/scala/akka/http/javadsl/server/Marshaller.scala

```diff
@@ -4,8 +4,108 @@
-trait Marshaller[T]
\ No newline at end of file
+import java.util.function
+import akka.http.scaladsl.marshalling
+import scala.concurrent.ExecutionContext
+import akka.http.javadsl.model.ContentType
+import akka.http.javadsl.model.MediaType
+import akka.http.scaladsl
+import akka.http.javadsl.model.HttpEntity
+import akka.http.scaladsl.marshalling._
+import akka.http.javadsl.model.HttpResponse
+import akka.http.javadsl.model.HttpRequest
+import akka.http.javadsl.model.RequestEntity
+import akka.util.ByteString
+import akka.http.scaladsl.model.FormData
+import akka.http.javadsl.model.StatusCode
+import akka.http.javadsl.model.HttpHeader
+import scala.collection.JavaConverters._
+import JavaScalaTypeEquivalence._
+
+object Marshaller {
+  def wrap[A,B](scalaMarshaller: marshalling.Marshaller[A,B]) = Marshaller()(scalaMarshaller)
+  
+  /** 
+   * Safe downcasting of the output type of the marshaller to a superclass. 
+   *  
+   * Marshaller is covariant in B, i.e. if B2 is a subclass of B1, 
+   * then Marshaller[X,B2] is OK to use where Marshaller[X,B1] is expected. 
+   */
+  def downcast[A, B1, B2 <: B1](m: Marshaller[A,B2]): Marshaller[A,B1] = m.asInstanceOf[Marshaller[A,B1]]
+
+  /** 
+   * Safe downcasting of the output type of the marshaller to a superclass. 
+   *  
+   * Marshaller is covariant in B, i.e. if B2 is a subclass of B1, 
+   * then Marshaller[X,B2] is OK to use where Marshaller[X,B1] is expected. 
+   */
+  def downcast[A, B1, B2 <: B1](m: Marshaller[A,B2], target: Class[B1]): Marshaller[A,B1] = m.asInstanceOf[Marshaller[A,B1]]
+  
+  def stringToEntity: Marshaller[String,RequestEntity] = wrap(marshalling.Marshaller.StringMarshaller)
+
+  def byteArrayToEntity: Marshaller[Array[Byte],RequestEntity] = wrap(marshalling.Marshaller.ByteArrayMarshaller)
+
+  def charArrayToEntity: Marshaller[Array[Char],RequestEntity] = wrap(marshalling.Marshaller.CharArrayMarshaller)
+  
+  def byteStringToEntity: Marshaller[ByteString,RequestEntity] = wrap(marshalling.Marshaller.ByteStringMarshaller)
+  
+  def fromDataToEntity: Marshaller[FormData,RequestEntity] = wrap(marshalling.Marshaller.FormDataMarshaller)
+  
+  def wrapEntity[A,C](f:function.BiFunction[ExecutionContext,C,A], m: Marshaller[A,_ <: RequestEntity], mediaType: MediaType): Marshaller[C,RequestEntity] = {
+    val scalaMarshaller = assumeScala(downcast(m, classOf[RequestEntity]).unwrap)
+    wrap(scalaMarshaller.wrapWithEC(mediaType) { ctx => c:C => f(ctx,c) } ) 
+  }
+
+  def wrapEntity[A,C](f:function.Function[C,A], m: Marshaller[A,_ <: RequestEntity], mediaType: MediaType): Marshaller[C,RequestEntity] = {
+    val scalaMarshaller = assumeScala(downcast(m, classOf[RequestEntity]).unwrap)
+    wrap(scalaMarshaller.wrap(mediaType)(f.apply)) 
+  }
+
+  def byteStringMarshaller(t: ContentType): Marshaller[ByteString, RequestEntity] = {
+    wrap(scaladsl.marshalling.Marshaller.byteStringMarshaller(t))
+  }
+  
+  def opaque[A,B](f: function.Function[A,B]): Marshaller[A, B] = {
```

Comment:
Actually I left out `strict` on purpose since that would require a Java API for the `Marshalling` trait. I haven't found a need for it... we've built a `Accept`-aware JSON+XML marshaller by simply doing `oneOf(xml, json)`. Should we pull adding `strict` into scope?

And isn't `combine` covered by `instance.map()` / `instance.compose()`, given that there are no implicits in Java anyways?